kbase-extension/static/kbase/js/util/jobLogViewer.js

```diff
@@ -687,143 +746,132 @@ define([
+                            throw new Error('Unknown log status ' + jobStatus);
+                    }
                     break;
-                case 'terminated':
-                    newState = {
-                        mode: 'canceled'
-                    };
+                case 'active':
+                    switch (jobStatus) {
+                        case 'queued':
+                            // this should not occur!
+                            break;
+                        case 'running':
+                            startAutoFetch();
+                            break;
+                        case 'completed':
+                            newState = {
+                                mode: 'complete',
+                            };
+                            break;
+                        case 'error':
+                            newState = {
+                                mode: 'error',
+                            };
+                            break;
+                        case 'terminated':
+                            newState = {
+                                mode: 'canceled',
+                            };
+                            break;
+                        default:
+                            console.error('Unknown log status', jobStatus, message);
+                            throw new Error('Unknown log status ' + jobStatus);
+                    }
                     break;
-                default:
-                    console.error('Unknown log status', jobStatus, message);
-                    throw new Error('Unknown log status ' + jobStatus);
-                }
-                break;
-            case 'complete':
-                switch (jobStatus) {
-                case 'completed':
-                    return;
-                default:
-                    // technically, an error, what to do?
-                    return;
-                }
-            case 'canceled':
-                switch (jobStatus) {
-                case 'terminated':
-                    return;
-                default:
+                case 'canceled':
+                    if (jobStatus === 'terminated') {
+                        return;
+                    }
                     console.error('Unexpected log status ' + jobStatus + ' for "canceled" state');
                     throw new Error('Unexpected log status ' + jobStatus + ' for "canceled" state');
-                }
-            case 'error':
-                switch (jobStatus) {
+                case 'complete':
+                    // N.b. if the jobStatus is not 'completed',
+                    // some sort of error has occurred
+                    return;
                 case 'error':
-                case 'suspend':
-                    // nothing to do;
+                    // N.b. if the jobStatus is not 'error' or 'suspend',
+                    // some sort of error has occurred
```

Comment:
this did have a `switch (jobStatus):` underneath it, but since the result of the switch was always `return`, regardless of the condition, I just removed it and put in this comment.